File: src/main/java/com/my/admin/app/commerce/model/document/Cash.java
package com.my.admin.app.commerce.model.document;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.ZonedDateTime;

@Document(collection = "cashes")
@Data
public class Cash implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    private String _id;

    private Turn turn;

    private User user;

    private CashStatus cashStatus;

    private Double startMoney;

    private Double endMoney;

    private ZonedDateTime startDate;

    private ZonedDateTime endDate;
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/model/document/Cash.kt
package com.bk.reactive.app.commerce.admin.my.model.document

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.time.ZonedDateTime

@Document(collection = "cashes")
data class Cash(
    @Id
    val id: String? = null,
    val turn: Turn,
    val user: User,
    val cashStatus: CashStatus,
    val startMoney: Double,
    val endMoney: Double? = null,
    var partialBalance: Double = 0.0,
    val startDate: ZonedDateTime,
    val endDate: ZonedDateTime? = null
)